sharding/indexHashedNodesCoordinator.go

```diff
@@ -179,6 +185,13 @@ func (ihgs *indexHashedNodesCoordinator) SetNodesPerShards(
 	nodesConfig.shardId = ihgs.computeShardForPublicKey(nodesConfig)
 	ihgs.nodesConfig[epoch] = nodesConfig
 
+	if updatePeersListAndIndex {
```

Comment:
is there any time where we are updating the PeerList but we do not want to UpdatePeersListAndIndex?

Reply:
yes. As discussed, when we first create the nodes coordinator we don't have the list index updater with ok values, but with the disabled one.

Reply:
I'm still not happy with the bool value, but it can be refactored in the future.